packages/main/src/Select.ts

```diff
@@ -286,10 +102,200 @@ const metadata = {
  * @public
  * @since 0.8.0
  */
-class Select extends UI5Element {
-	static get metadata() {
-		return metadata;
-	}
+@customElement("ui5-select")
+@languageAware
+/**
+ * Fired when the selected option changes.
+ *
+ * @event sap.ui.webc.main.Select#change
+ * @param {HTMLElement} selectedOption the selected option.
+ * @public
+ */
+@event("change", {
+	detail: {
```

Comment:
We also have this convention for creating a type for any public event with detail - type SelectChangeEventDetail, you can look into other components, there are plenty of examples. The idea is to describe the event detail and later when you fire the event via fireEvent to use this type, f.e:

```js
this.fireEvent<SelectChangeEventDetail>("change", {
			selectedOption: Option
});
```